src/_pyinstaller_hooks_contrib/hooks/pre_safe_import_module/hook-win32com.py

```diff
@@ -22,13 +22,25 @@
 
 import os
 
-from PyInstaller.utils.hooks import logger, get_module_file_attribute
+from PyInstaller.utils.hooks import logger, get_module_file_attribute, \
+                                    exec_statement
 from PyInstaller.compat import is_win, is_cygwin
 
 
 def pre_safe_import_module(api):
     if not (is_win or is_cygwin):
         return
+    is_available = exec_statement(
+        """
+        try:
+            from win32com import __file__
+            print(__file__)
+        except Exception:
+            pass
+        """).strip()
+    if not is_available:
+        logger.debug('win32com: module not not available')
+        return  # win32com unavailable
     win32com_dir = os.path.dirname(get_module_file_attribute('win32com'))
```

Comment:
Is there a reason you chose to keep the `get_module_file_attribute()` rather than using the output from the `exec_statement()` call?

Reply:
Ah, I didn't pay enough attention to what `get_module_file_attribute` actually does. But you're right, it is redundant now.